include/cute/atom/copy_traits_xe.hpp

```diff
@@ -1,80 +1,279 @@
+  }
+};
+
+template <class GTensor> struct Copy_Traits<XE_2D_U32X2X1_N, GTensor> {
+  using ThrID = Layout<_16>;
+  // Map from (src-thr,src-val) to bit
+  using SrcLayout = Layout<Shape<_16, _64>, Stride<_0, _1>>;
+  // Map from (dst-thr,dst-val) to bit
+  using DstLayout =
+      Layout<Shape<_16, Shape<_16, _32>>, Stride<_32, Stride<_512, _1>>>;
+  // Reference map from (thr,val) to bit
+  using RefLayout = DstLayout;
+  // 32 bits register file
+  using CopyInternalType = uint;
+
+  GTensor tensor;
+
+  template <class TS, class SLayout, class TD, class DLayout>
+  CUTE_HOST_DEVICE friend constexpr void copy_unpack(
+      Copy_Traits const &traits,
+      Tensor<ViewEngine<ArithmeticTupleIterator<TS>>, SLayout> const &src,
+      Tensor<TD, DLayout> &dst) {
+    static_assert(is_rmem<TD>::value);
+    int H = size<0>(traits.tensor);
+    // int W = size<1>(traits.tensor) * sizeof(typename
+    // decltype(traits.tensor)::engine_type::value_type);
+    int W = size<1>(traits.tensor) *
+            sizeof(CopyInternalType); // TODO: inconsistent to give the size in
+                                      // elements but use vector for copy
+    auto [y, x] = src.data().coord_;
+    XE_2D_U32X2X1_N::copy(traits.tensor.data().get(), W, H, W, int2_{x, y},
+                          &*dst.data());
+  }
+};
+
+template <class GTensor> struct Copy_Traits<XE_2D_U16X2X1_N, GTensor> {
+  // // using ThrID   = Layout<_16>; //TODO: I think it should be 16 (copy is
+  // per subgroup) - but static_assert fails
+  using ThrID = Layout<_16>;
+  // Map from (src-thr,src-val) to bit
+  using SrcLayout = Layout<Shape<_16, _64>, Stride<_0, _1>>;
+  // Map from (dst-thr,dst-val) to bit
+  using DstLayout =
+      Layout<Shape<_16, Shape<_16, _32>>, Stride<_32, Stride<_512, _1>>>;
+  // Reference map from (thr,val) to bit
+  using RefLayout = DstLayout;
+  // 32 bits register file
+  using CopyInternalType = uint;
+
+  GTensor tensor;
+
+  template <class TS, class SLayout, class TD, class DLayout>
+  CUTE_HOST_DEVICE friend constexpr void copy_unpack(
+      Copy_Traits const &traits,
+      Tensor<ViewEngine<ArithmeticTupleIterator<TS>>, SLayout> const &src,
+      Tensor<TD, DLayout> &dst) {
+    static_assert(is_rmem<TD>::value);
+    int H = size<0>(traits.tensor);
+    // int W = size<1>(traits.tensor) * sizeof(typename
+    // decltype(traits.tensor)::engine_type::value_type);
+    int W = size<1>(traits.tensor) *
+            sizeof(CopyInternalType); // TODO: inconsistent to give the size in
+                                      // elements but use vector for copy
+    auto [y, x] = src.data().coord_;
+    XE_2D_U16X2X1_N::copy(traits.tensor.data().get(), W, H, W, int2_{x, y},
+                          &*dst.data());
+  }
+};
+
+template <class GEngine, class GLayout>
+auto make_xe_2d_copy(Tensor<GEngine, GLayout> gtensor) {
+  using GTensor = Tensor<GEngine, GLayout>;
+  using Traits = Copy_Traits<XE_2D_LOAD, GTensor>;
```

Comment:
I don't think we want `XE_2D_LOAD` hard-coded here. Either we want the copy to be an argument (as it was before), or we compute the correct load tile size as part of this function.